web/assets/js/GoogleAnalyticsItem.js

```diff
@@ -0,0 +1,12 @@
+class GoogleAnalyticsItem {
+
+    constructor(hitType = 'event', eventCategory, eventAction, eventLabel = null, transport = {}) {
```

Comment:
Params with default values should go last. Also I would suggest to pass an object to constructor instead of a list of params, for better readability and to make it less prone for breaking changes in the future.
Having 5 params in one function call is something undesired.

Reply:
What's more. I don't see any value in creating a class that stores only values and doesn't do anything else. Pure object would work better.

Reply:
In this case GoogleAnalyticsItem class is something like Data Transfer Object. I added it because it is not a good practice to call external methods directly, in our case ga(hitType, eventCategory, eventAction, eventLabel, transport). The main value of this type of class is that you don't need to know what's inside external method or class and you don't worry about situation when implementation will change. You pass required attributes and inside some service the right method preparing data to send. Owing to the fact we have better control on our code.
I fully agree with you that a better way is to pass an object to constructor instead of a list of params, but I think that in this case will be more readable to pass this attributes.

Reply:
But this class doesn't implement anything but the constructor where attributes passed in the constructor are accessible as class instance properties. I see no difference between class and object in this case. There are no extra methods defined, just a constructor that does nothing else but acts like an object. That's the point of my comment. You defined a class for wrong purpose. @dew326 any opinions on that?

Reply:
I agree with @sunpietro. This is should be an object or map, not a class.